packages/decoder/lib/Web3Adapter.ts

```diff
@@ -0,0 +1,50 @@
+import Web3Eth from "web3-eth";
+type PastLogsOptions = {
+  toBlock?: string | number;
+  fromBlock?: string | number;
+  address?: string | string[];
+}
+import type BN from "bn.js";
+
+export class Web3Adapter {
+  public provider: any;
+  private eth: any;
+
+  constructor (provider: any) {
+    this.provider = provider;
+    // @ts-ignore - seems like the types are out of sync with the implementation
+    this.eth = new Web3Eth(provider);
+  }
+
+  public async getCode (address: string, block: string | number) {
+    return await this.eth.getCode(address, block);
+  }
+
+  public async getBlock (block: string | number) {
+    return await this.eth.getBlock(block);
+  }
+
+  public async getPastLogs ({ address, fromBlock, toBlock }: PastLogsOptions): Promise<any[]> {
+    return await this.eth.getPastLogs({ address, fromBlock, toBlock });
+  }
+
+  public async getNetworkId () {
+    return (await this.eth.net.getId()).toString();
+  }
+
+  public async getBlockNumber (): Promise<number> {
+    return await this.eth.getBlockNumber();
+  }
+
+  public async getBalance (address: string, blockNumber: any) {
```

Comment:
I get why this `any` might be useful, but I don't like it here in this internal adapter. It's pretty bad if the decoder and this adapter can't agree on a type more well-defined than `any`!

Reply:
(also what's the return value?)